test/ComposerGpgVerifyTest/VerifyTest.php

```diff
@@ -0,0 +1,445 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ComposerGpgVerifyTest;
+
+use Composer\Composer;
+use Composer\Config;
+use Composer\Script\Event;
+use Composer\Script\ScriptEvents;
+use ComposerGpgVerify\Verify;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Process\Process;
+
+/**
+ * @covers \ComposerGpgVerify\Verify
+ */
+final class VerifyTest extends TestCase
```

Comment:
why checking HEAD of the repo ? what is important is that the version being installed is trusted, which may not be latest commit of the repo. If your project has a good version in the lock file, it should not be forbidden to install its deps, even though the upstream repo may have bad commits after that in the history